sdk/python/kfp_tekton/compiler/_op_to_template.py

```diff
@@ -133,20 +133,44 @@ def _op_to_template(op: BaseOp):
     outputs_dict = _outputs_to_json(op, processed_op.outputs, param_outputs, output_artifacts)
     if outputs_dict:
```

Comment:
NIT: I think it would be handy to annotate the code here with comments that include samples of the generated Tekton YAML - however the test data already helps a lot!